odbc/src/api/functions.rs

```diff
@@ -414,7 +414,38 @@ pub unsafe extern "C" fn SQLBulkOperations(
 #[named]
 #[no_mangle]
 pub unsafe extern "C" fn SQLCancel(statement_handle: HStmt) -> SqlReturn {
-    unimpl!(statement_handle);
+    panic_safe_exec_keep_diagnostics!(
```

Comment:
I thought we were going to drop the mongo_statement for some reason.
I checked the code, I am not sure how we re-use statements (or if we actually can, I'd have to dig a little more), but if we are in the middle of retrieving data and cancel is called, we should close the cursor or drop the mongo_statement somehow if it's possible.

Reply:
I agree this is a bit unclear, but the case that motivates my uncertainty would be: if you issue a query using SQLPrepare followed by SQLExecute, then call SQLCancel, should you be able to call SQLExecute again? If the answer is yes, I believe we need to keep the `mongo_statement` in tact. 

I was thinking the answer to this might be yes, since other behaviors we don't support for SQLCancel behave this way -- namely [cancelling functions that need data](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcancel-function?view=sql-server-ver16#canceling-functions-that-need-data). So My thought was to just allow the program to behave as if execution failed and let the thread calling SQLExecute figure out how to proceed

Reply:
Yes, that seems like the right thing to do.
If found this in the spec, which corroborate your thinking. "To close a cursor, an application should call SQLCloseCursor, not SQLCancel." 👍